### DISCUSSION TITLE
Make execution functions payable in the standard

### DISCUSSION THREAD

**Discussion by filmakarov** [erc7579/erc7579-implementation#2] [Ideas]
## Idea

There are use cases, that require passing value from the Executor to the SA along with `executeFromExecutor` and `executeBatchFromExecutor` calls.
For example, a module that enables perp trading features with the MSA.
In such a use case, to keep the interface consistent between perp DEX and the module and avoid extra gas overhead, the above mentioned methods should be payable.

They [are](https://github.com/erc7579/erc7579-implementation/blob/7369abdadc0ba2dd87321c6e4427d6c98a8df2f5/src/MSABase.sol#L86-L113) currently payable in the reference implementation.

I propose marking them (along with other execution methods) in the [ERC description](https://github.com/ethereum/ERCs/pull/163/files) as well.

**Comment by zeroknots** [comment]
I think this is a pretty good idea. we should do this if there is no vetos :D
